util/cpuprofile/cpuprofile.go

```diff
@@ -0,0 +1,219 @@
+type ProfileData struct {
+	Data  *bytes.Buffer
+	Begin time.Time
+	End   time.Time
+	Error error
+}
+
+// StartCPUProfiler uses to start to run the global parallelCPUProfiler.
+func StartCPUProfiler() error {
+	return globalCPUProfiler.start()
+}
+
+// StopCPUProfiler uses to stop the global parallelCPUProfiler.
+func StopCPUProfiler() {
+	globalCPUProfiler.stop()
+}
+
+// Register register a ProfileConsumer into the global CPU profiler.
+// Normally, the registered ProfileConsumer will receive the cpu profile data per second.
+// If the ProfileConsumer (channel) is full, the latest cpu profile data will not be sent to it.
+// This function is thread-safe.
+func Register(ch ProfileConsumer) {
+	globalCPUProfiler.register(ch)
+}
+
+// Unregister unregister a ProfileConsumer from the global CPU profiler.
+// The unregistered ProfileConsumer won't receive the cpu profile data any more.
+// This function is thread-safe.
+func Unregister(ch ProfileConsumer) {
+	globalCPUProfiler.unregister(ch)
+}
+
+// parallelCPUProfiler is a cpu profiler.
+// With parallelCPUProfiler, it is possible to have multiple profile consumer at the same time.
+// WARN: Only one running parallelCPUProfiler is allowed in the process, otherwise some profiler may profiling fail.
+type parallelCPUProfiler struct {
+	sync.Mutex
+	cs             map[ProfileConsumer]struct{}
+	notifyRegister chan struct{}
+
+	// some data cache for profiling.
+	data         *ProfileData
+	lastDataSize int
+
+	started bool
+	ctx     context.Context
+	cancel  context.CancelFunc
+	wg      sync.WaitGroup
+}
+
+// newParallelCPUProfiler crate a new parallelCPUProfiler.
+func newParallelCPUProfiler() *parallelCPUProfiler {
+	return &parallelCPUProfiler{
+		cs:             make(map[ProfileConsumer]struct{}),
+		notifyRegister: make(chan struct{}),
+	}
+}
+
+var (
+	errProfilerAlreadyStarted = errors.New("parallelCPUProfiler is already started")
+)
+
+func (p *parallelCPUProfiler) start() error {
+	p.Lock()
+	defer p.Unlock()
+	if p.started {
+		return errProfilerAlreadyStarted
+	}
+
+	p.started = true
+	p.ctx, p.cancel = context.WithCancel(context.Background())
+	p.wg.Add(1)
+	go util.WithRecovery(p.profilingLoop, nil)
+
+	logutil.BgLogger().Info("parallel cpu profiler started")
+	return nil
+}
+
+func (p *parallelCPUProfiler) stop() {
+	p.Lock()
+	defer p.Unlock()
+	if !p.started {
+		return
+	}
+	p.started = false
+	if p.cancel != nil {
+		p.cancel()
+	}
+	p.wg.Wait()
+	logutil.BgLogger().Info("parallel cpu profiler stopped")
+}
+
+func (p *parallelCPUProfiler) register(ch ProfileConsumer) {
+	if ch == nil {
+		return
+	}
+	p.Lock()
+	p.cs[ch] = struct{}{}
+	p.Unlock()
+
+	select {
+	case p.notifyRegister <- struct{}{}:
+	default:
+	}
+}
+
+func (p *parallelCPUProfiler) unregister(ch ProfileConsumer) {
+	if ch == nil {
+		return
+	}
+	p.Lock()
+	delete(p.cs, ch)
+	p.Unlock()
+}
+
+func (p *parallelCPUProfiler) profilingLoop() {
```

Comment:
"Loop" is duplicated with "Cycle". They are same meanings. Please change names to clearly indicate their different roles.